website/content/en/docs/kubebuilder/operator-scope.md

```diff
@@ -0,0 +1,268 @@
+```
+
+And then, run `make manifests` to update `config/rbac/role.yaml`:
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+...
+```
+
+<!-- todo(camilamacedo86): remove the note or improve the above info when we have further info in the issue tracked  -->     
+**NOTE**: Note that the above steps are not updating the `ClusterRoleBinding` to `RoleBinding` in `config/rbac/role_binding.yaml` which need to be done manually and shows a bug. More info: https://github.com/kubernetes-sigs/kubebuilder/issues/1496.
+
+## Using environment variables for Namespace 
+
+Instead of having the specific Namespace(s) hard-coded in the `main.go` file a good practice is to use environment 
+variables to allow the restrictive configurations
+
+### Configuring Namespace scoped operators
+
+- Add a helper function in the `main.go` file:
+
+```go
+// getWatchNamespace returns the Namespace the operator should be watching for changes
+func getWatchNamespace() (string, error) {
+    // WatchNamespaceEnvVar is the constant for env variable WATCH_NAMESPACE
+    // which is the Namespace where the watch activity happens.
+    // this value is empty if the operator is running with clusterScope.
+    var watchNamespaceEnvVar = "WATCH_NAMESPACE"
+    
+    ns, found := os.LookupEnv(watchNamespaceEnvVar)
+    if !found {
+        return "", fmt.Errorf("%s must be set", watchNamespaceEnvVar)
+    }
+    return ns, nil
+}
+```
+
+- Use the environment variable value: 
+
+```go
+...
+watchNamespace, err := getWatchNamespace()
+if err != nil {
+    setupLog.Error(err, "unable to get WatchNamespace, " +
+        "the manager will watch and manage resources in all cluster")
+}
+
+mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+    Scheme:             scheme,
+    MetricsBindAddress: metricsAddr,
+    Port:               9443,
+    LeaderElection:     enableLeaderElection,
+    LeaderElectionID:   "f1c5ece8.example.com",
+    Namespace:          watchNamespace, // namespaced-scope when the value is not an empty string 
+})
+...
+```
+
+- Define the environment variable in the `config/manager/manager.yaml`:
+
+```yaml
+spec:
+  containers:
+  - command:
+    - /manager
+    args:
+    - --enable-leader-election
+    image: controller:latest
+    name: manager
+    resources:
+      limits:
+        cpu: 100m
+        memory: 30Mi
+      requests:
+        cpu: 100m
+        memory: 20Mi
+    env:
+      - name: WATCH_NAMESPACE
+        valueFrom:
+          fieldRef:
+            fieldPath: metadata.namespace 
+  terminationGracePeriodSeconds: 10
+``` 
+
+**NOTE** The above will set as value of `WATCH_NAMESPACE` the namespace where the operator will be deployed. 
+
+### Configuring cluster-scoped operators with MultiNamespacedCacheBuilder
+
+- Add a helper function in the `main.go` file :
+
+```go
+// getWatchNamespace returns the namespace the operator should be watching for changes
+func getWatchNamespace() (string, error) {
```

Comment:
Ditto above.

Reply:
^ditto

Reply:
We have not k8sutil in KB layout and we do not know if we will keep it.